binary_wheel_builder/api/build.py

```diff
@@ -106,30 +108,43 @@ def build_wheel(wheel_meta: Wheel, dist_folder: Path, worker_count: int = 1) ->
     :return: Yields for each generated platform wheel
     """
     dist_folder.mkdir(exist_ok=True)
-    with concurrent.futures.ThreadPoolExecutor(max_workers=worker_count) as executor:
+  
+    worker_count = worker_count or os.cpu_count()
```

Comment:
note: The worker count was chosen deliberately

This is intended to run mostly in CI environments with [cgroups ](https://blog.nginx.org/blog/what-are-namespaces-cgroups-how-do-they-work) defined, which leads to false postives of CPU counts. To avoid weird OOM kill bugs, I set the worker count to a reasonable default with freedom to pick a custom limit.

Reply:
Good. I didn´t think about that possible bugs